generate-wrapper.py

```diff
@@ -59,43 +101,60 @@ def visit_FuncDecl(self, node):
 
        if node.args:
            for p in node.args.params:
-               args.append(typename(p.type))
+               args.append((p.name, typename(p.type)))
 
        if name == '__routine':
+           print(f'function not handled: {name}')
            return # ??
 
        if 'replacement' in name:
+           print(f'function not handled: {name}')
            return # ??
 
        if 'delete_callback' in name:
+           print(f'function not handled: {name}')
+           self.types_result += f'export type {name} = (data: pointer) => void;\n'
            return # ??
 
 
        if 'table_function' in name:
+           print(f'function not handled: {name}')
            return # TODO
 
        if 'arrow' in name:
+           print(f'function not handled: {name}')
            return # TODO
 
        if name in deprecated_functions:
            return
 
        #print(f"{ret} {name} ({', '.join(args)})")
-       print(f"{name}")
+       #print(f"{name}")
 
        n_args = len(args)
-       args.append(name)
+
+       fwrap_args = list(map(lambda arg: arg[1], args)) + [name]
```

Comment:
Rather than modify `args`, which I need for the TS generation, I'm creating a copy for the C++ generation. The C++ needs the arg types (not the names), plus the name of the function.